drake/systems/plants/joints/RevoluteJoint.h

```diff
@@ -2,8 +2,9 @@
 #define REVOLUTEJOINT_H_
 
 #include "FixedAxisOneDoFJoint.h"
+#include <Eigen/Geometry>
 
-class DLLEXPORT_DRAKEJOINT RevoluteJoint: public FixedAxisOneDoFJoint
+class DLLEXPORT_DRAKEJOINT RevoluteJoint: public FixedAxisOneDoFJoint<RevoluteJoint>
```

Comment:
regarding #1273 , i see that this second-level derived class doesn't overload any of the methods from FixedAxisOneDoFJoint.  I still think that you would have trouble doing that with this pattern.  E.g., what if you wanted to overload v2qdot here?  the compiler would complain that you're shadowing a non-virtual method.  no?